src/ios/registerNativeModule.js

```diff
@@ -50,6 +54,74 @@ const addProjectToLibraries = (libraries, file) => {
 };
 
 /**
+ * Given folder, it returns an array of all header files
+ * inside it, ignoring node_modules and examples
+ */
+const getHeadersInFolder = (folder) =>
+  glob
+    .sync('**/*.h', {
+      cwd: folder,
+      nodir: true,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    .map(file => path.join(folder, file));
+
+/**
+ * Given an array of headers
+ * it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+const getHeaderSearchPath = (from, headers) => {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(from, directories[0])}"`;
+  }
+
+  const directory = directories.reduce((topMostDir, currentDir) => {
+    return currentDir.split(path.sep) < topMostDir.split(path.sep)
+      ? currentDir
+      : topMostDir;
+  });
+
+  return `"$(SRCROOT)/${path.relative(from, directory)}/**"`;
+};
```

Comment:
Thanks to that method, it now properly handles react-native-code-push by returning:
`"$(SRCROOT)/../node_modules/react-native-code-push"` for it and my fake library that has nested `src` folder (common-case) by returning:
`"$(SRCROOT)/../node_modules/react-native-linear-gradient/src/**",`
because there are src/module/module.h and src/moduleb/moduleb.h